bdk-ffi/src/lib.rs

```diff
@@ -47,17 +47,20 @@ pub struct ScriptAmount {
 
 /// A derived address and the index it was found at.
 pub struct AddressInfo {
-    /// Child index of this address
+    /// Child index of this address.
     pub index: u32,
-    /// Address
-    pub address: String,
+    /// Address.
+    pub address: Arc<Address>,
+    /// Type of keychain.
+    pub keychain: KeychainKind,
 }
 
 impl From<BdkAddressInfo> for AddressInfo {
-    fn from(x: bdk::wallet::AddressInfo) -> Self {
+    fn from(x: BdkAddressInfo) -> Self {
```

Comment:
nit: can we use something more descriptive instead of `x`, something like `ai`:
```suggestion
    fn from(ai: BdkAddressInfo) -> Self {
```